core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnRemove.java

```diff
@@ -0,0 +1,101 @@
+ * INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES, ARISING OUT OF, RESULTING FROM,
+ * OR IN ANY WAY CONNECTED WITH THIS SOFTWARE, WHETHER OR NOT BASED UPON WARRANTY,
+ * CONTRACT, TORT, OR OTHERWISE, WHETHER OR NOT INJURY WAS SUSTAINED BY PERSONS OR
+ * PROPERTY OR OTHERWISE, AND WHETHER OR NOT LOSS WAS SUSTAINED FROM, OR AROSE OUT
+ * OF THE RESULTS OF, OR USE OF, THE SOFTWARE OR SERVICES PROVIDED HEREUNDER.
+ */
+
+package gov.nist.secauto.metaschema.core.metapath.function.library;
+
+import gov.nist.secauto.metaschema.core.metapath.DynamicContext;
+import gov.nist.secauto.metaschema.core.metapath.ISequence;
+import gov.nist.secauto.metaschema.core.metapath.MetapathConstants;
+import gov.nist.secauto.metaschema.core.metapath.function.FunctionUtils;
+import gov.nist.secauto.metaschema.core.metapath.function.IArgument;
+import gov.nist.secauto.metaschema.core.metapath.function.IFunction;
+import gov.nist.secauto.metaschema.core.metapath.item.IItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IIntegerItem;
+import gov.nist.secauto.metaschema.core.util.ObjectUtils;
+
+import java.util.Collections;
+import java.util.List;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Implements <a href=
+ * "https://www.w3.org/TR/xpath-functions-31/#func-exists">fn:exists</a>.
+ */
+public final class FnRemove {
+  @NonNull
+  static final IFunction SIGNATURE_ONE_ARG = IFunction.builder()
+      .name("remove")
+      .namespace(MetapathConstants.NS_METAPATH_FUNCTIONS)
+      .deterministic()
+      .contextIndependent()
+      .focusIndependent()
+      .argument(IArgument.builder()
+          .name("target")
+          .type(IItem.class)
+          .zeroOrMore()
+          .build())
+      .argument(IArgument.builder()
+          .name("position")
+          .type(IIntegerItem.class)
+          .zeroOrMore()
+          .build())
+      .returnType(IItem.class)
+      .returnZeroOrMore()
+      .functionHandler(FnRemove::execute)
+      .build();
+
+  private FnRemove() {
+    // disable construction
+  }
+
+  @SuppressWarnings("unused")
+  @NonNull
+  private static ISequence<?> execute(@NonNull IFunction function,
+      @NonNull List<ISequence<?>> arguments,
+      @NonNull DynamicContext dynamicContext,
+      IItem focus) {
+    ISequence<?> target = FunctionUtils.asType(ObjectUtils.requireNonNull(arguments.get(0)));
+    return ISequence.of(fnRemove(target));
+  }
+
+  /**
+   * Identify if there is at least one item in the {@code sequence}.
+   *
+   * @param sequence
+   *          the sequence to check
+   * @return {@code true} if the sequence contains at least one item, or
+   *         {@code false} otherwise
+   */
+  public static List<? extends IItem> fnRemove(List<? extends IItem> target, IIntegerItem position) {
+    if (target.size() <= 1) {
+      return target;
+    } //add else if target.size()< position?
+    
+    target.remove(position); 
+    return target;
```

Comment:
This logic is not the same as what is described in the [spec](https://www.w3.org/TR/xpath-functions-31/#func-remove).

> The function returns a sequence consisting of all items of $target whose index is less than $position, followed by all items of $target whose index is greater than $position. 

Specifically, the note about the position:

> If $position is less than 1 or greater than the number of items in $target, $target is returned. 

> If $target is the empty sequence, the empty sequence is returned. 

Also, a copy of target should be made, since there is no gaurantee that the target is mutable. The logic here should be similar to `insert-before` relative to the positioning.

Reply:
Resolved with https://github.com/aj-stein-nist/metaschema-java/commit/63be2a97a8558900a52b58221936b98b632eccd0.